be/src/io/cache_input_stream.cpp

```diff
@@ -252,16 +253,15 @@ Status CacheInputStream::_populate_to_cache(const int64_t offset, const int64_t
             }
         }
         Status r = _cache->write_buffer(_cache_key, write_offset_cursor, write_size, src_cursor, &options);
-        if (r.ok()) {
+        if (r.ok() || r.is_already_exist()) {
             _stats.write_cache_count += 1;
```

Comment:
if `is_already_exist`, shall we record write statistics?

Reply:
> if `is_already_exist`, shall we record write statistics?

I think it's ok, we can treat it as an optimization for writing same object repeatedly with `overwrite=false`. And if we write item with `overwrite=true`, the underly will replace the old item and write a new one. From the user, no matter which case above, the target has been written to cache, so it can be a success operations.